File: 1441.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        ans=list()
        compare=list()
        targetind=0
        for i in range(n):
            compare.append(i+1)
            ans.append("Push")
            if target[targetind]!=compare[-1]:
                ans.append("Pop")
                compare.pop()
            else:
                targetind+=1
            if targetind>=len(target):
                break
        return ans
            
            

File: 1441.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> answer;
        vector<int> compare;
        int targetind=0;
        
        for(int i=0;i<n;i++)
        {
            compare.push_back(i+1);
            answer.push_back("Push");
            if(compare.back()!=target[targetind])
            {
                compare.pop_back();
                answer.push_back("Pop");
            }
            else
            {
                targetind++;
            }
            if(targetind==size(target))
            {
                break;
            }
        }
        return answer;
    }
};
